numerics/polynomial.hpp

```diff
@@ -38,9 +38,9 @@ class Polynomial {
   friend constexpr bool operator!=(Polynomial const& left,
                                    Polynomial const& right) = default;
 
-  virtual Value operator()(Argument const& argument) const = 0;
-  virtual Derivative<Value, Argument> EvaluateDerivative(
-      Argument const& argument) const = 0;
+  virtual Value __vectorcall operator()(Argument argument) const = 0;
```

Comment:
Clang seems to pass things in registers with its default calling convention, so let’s make __vectorcall conditional on MSVC.

See https://godbolt.org/z/MqTP8vcK8.